### PR TITLE
DOC: Adding docs contribution guidelines

### LINKED ISSUE #16603: ENH, DOC: Add policy on typo and small docs fixes
### Is your feature request related to a problem? Please describe.

Continuing the discussion from #16578, and summarizing some of the comments:

- It would be ideal to accumulate doc changes rather then send typo fixes one by one if possible, and also try to use the correct commands to skip CI
- The easiest fixes to accept are those where the original version is clearly and unambiguously wrong; changes that require subtle editorial judgement are probably best avoided. (But note that this is not about updating documentation to fix confusing statements or otherwise deal with documentation problems reported by users.)
- One can imagine, however, cases where it is not clear that a "fix" should be made, e.g.:
	- Attempting to "fix" all the uses (or lack thereof) of the [Oxford comma](https://en.wikipedia.org/wiki/Serial_comma).
	- The sentence that I wrote above that begins "A problem with" has an issue (or issues!) that some might be inclined to fix.
	- Cases where the correctness of common usage is evolving, e.g. ["comprised of"](https://www.merriam-webster.com/words-at-play/can-you-use-comprised-of-grammar) (currently only occurring in a few comments in the boost submodule, so not an actual issue in SciPy--phew).
- [NumPy has a bit on this](https://numpy.org/doc/stable/dev/howto-docs.html#contributing-fixes), and we can probably take the opportunity to get similar wording for the policies of both projects

### Describe the solution you'd like.

Write a policy similar the the NumPy one about submitting smaller typo and documentation fixes.

An additional suggestion is to have a specific DOC issue template linking to that policy.

### Describe alternatives you've considered.

_No response_

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by melissawm**
#### Reference issue
Fixes #16603

#### What does this implement/fix?
Expands on the existing guidelines, moving them to a separate doc. Heavily inspired by the [NumPy documentation guidelines](https://numpy.org/doc/stable/dev/howto-docs.html).

### CODE REVIEW

File: doc/source/dev/contributor/rendering_documentation.rst [hunk 1]
```diff
@@ -69,6 +109,114 @@ on the cloud.
...
+            ----------
+            x : float
+                `x` must be nonnegative.
+
+    No::
+
+            Parameters
+            ----------
+            x : float
+                `x` should be nonnegative.
```
Comment (tylerjereddy) [inline doc/source/dev/contributor/rendering_documentation.rst:136]: Interesting, `git grep -E -A 4 "Parameters" | grep -E -i "should" | wc -l` returns about 50 results.

**Review by tylerjereddy — APPROVED** [review]
I read through the full diff and it all seems reasonable enough to me. I didn't check the rendered version, though I see the doc build is indeed passing.

**Comment by rgommers** [comment]
This looks like a good place to explain that docs-only PRs only should get built on CircleCI, and to achieve that the commit message should contain `[skip actions][skip azp]`.

**Review by tupui — APPROVED** [review]
Thanks @melissawm. I particularly like the note about typos and skipping of the CI.

**Review by rgommers — APPROVED** [review]
I had a more detailed look now, it looks great. Just pushed a 2 line fix for some formatting. In it goes, thanks @melissawm!

**Pull request merged**